### PR TITLE
wallet, tests: Avoid stringop-overflow warning in PollutePubKey

### PR DESCRIPTION
**Pull request by achow101**
Fixes  #30114

### CODE REVIEW

File: src/wallet/test/wallet_tests.cpp [hunk 1]
```diff
@@ -499,8 +499,10 @@ static void TestWatchOnlyPubKey(LegacyScriptPubKeyMan* spk_man, const CPubKey& a
 // Cryptographically invalidate a PubKey whilst keeping length and first byte
 static void PollutePubKey(CPubKey& pubkey)
 {
-    std::vector<unsigned char> pubkey_raw(pubkey.begin(), pubkey.end());
-    std::fill(pubkey_raw.begin()+1, pubkey_raw.end(), 0);
+    assert(pubkey.IsValid());
```
Comment (laanwj) [inline src/wallet/test/wallet_tests.cpp:502]: Maybe directly assert `pubkey.size() >= 1` to clarify.

Comment (achow101) [reply src/wallet/test/wallet_tests.cpp]: Done

**Review by theStack — APPROVED** [review]
utACK 2289d4524053ab71c0d9133987cb36412797c1a2

**Review by laanwj — APPROVED** [review]
ACK 2289d4524053ab71c0d9133987cb36412797c1a2

**Pull request merged**